libs/agno/agno/agent/agent.py

```diff
@@ -5918,6 +5970,60 @@ def get_session_state(self, session_id: Optional[str] = None) -> Dict[str, Any]:
             raise Exception("Session not found")
         return session.session_data.get("session_state", {}) if session.session_data is not None else {}
 
+    def update_session_state(self, session_state_updates: Dict[str, Any], session_id: Optional[str] = None) -> str:
+        """
+        Update the session state for the given session ID and user ID.
+        Args:
+            session_state_updates: The updates to apply to the session state. Should be a dictionary of key-value pairs.
+            session_id: The session ID to update. If not provided, the current cached session ID is used.
+        Returns:
+            dict: The updated session state.
+        """
+        session_id = session_id or self.session_id
+        if session_id is None:
+            raise Exception("Session ID is not set")
+        session = self.get_session(session_id=session_id)  # type: ignore
+        if session is None:
+            raise Exception("Session not found")
+
+        if session.session_data is not None and "session_state" not in session.session_data:
+            session.session_data["session_state"] = {}
+
+        for key, value in session_state_updates.items():
+            session.session_data["session_state"][key] = value
```

Comment:
Should this be a hard overwrite like this or a dict merge?

Thinking about cases like our shopping list example. Maybe doing `update_session_state(session_state_updates: {"shopping_list": ["apple"]}` I would want to "add apple", not "overwrite list in favor of ["apple"]". Could be done adding a `merge_with_current_session_state: bool`? idk if relevant.